subprojects/core/src/main/java/org/gradle/plugin/management/internal/PluginRequestsSerializer.java

```diff
@@ -49,7 +50,7 @@ public void write(Encoder encoder, PluginRequests requests) throws Exception {
             encoder.writeString(request.getId().getId());
             encoder.writeNullableString(request.getVersion());
             encoder.writeBoolean(request.isApply());
-            encoder.writeSmallInt(request.getLineNumber());
+            encoder.writeNullableString(request.getLineNumber() == null ? null : request.getLineNumber().toString());
```

Comment:
⭕️ Why convert it to a String? -1 would be a reasonable placeholder.

Reply:
I had a more generic approach in mind that can also handle negative numbers which would we could abstract into a serializer method in the long run. In this specific case you are right. A line number of -1 could represent a null value.